packages/bottom-tabs/src/views/BottomTabBar.tsx

```diff
@@ -249,18 +260,87 @@ export default function BottomTabBar({
     descriptors,
     dimensions,
     layout,
+    tabBarPosition,
   });
 
   const tabBarBackgroundElement = tabBarBackground?.();
 
+  const tabBarPositionMap: TabBarPositionMapConfig = {
+    right: {
+      tabBarItem: {
+        height:
+          Platform.OS === ('ios' || 'android')
+            ? tabBarHeight / 1.5
+            : DEFAULT_TABBAR_HEIGHT,
+        marginTop: COMPACT_TABBAR_HEIGHT / 2,
```

Comment:
The height of tab items is unrelated to the tab bar height. They shouldn't be calculated based on what the tab bar height is. Ideally, the height of each item would be dynamic based on the size of the icon and label and not hard coded.

The margins should be hard-coded to a constant which is not dependent on tab bar height. I'd prefer them to be `marginVertical` instead of `marginTop` so they are spread between each item.

Reply:
Removing the height and using flex makes the tabBarItem looks like this, this is because the tabBarItem has no height and its element height is undefined

```js
   tabBarItem: {
        flex: 0,
        borderWidth: 1,
        marginVertical: COMPACT_TABITEM_GAP,
      },
```

![Screenshot 2022-11-16 at 10 19 11 AM](https://user-images.githubusercontent.com/33870502/202140697-4b11e336-a470-4f1a-a08a-eeba073c6ea3.png)

```js
   tabBarItem: {
        flex: 1,
        borderWidth: 1,
        marginVertical: COMPACT_TABITEM_GAP,
      },
```

<img width="960" alt="Screenshot 2022-11-16 at 10 18 58 AM" src="https://user-images.githubusercontent.com/33870502/202140729-8bf0a142-f48e-4215-8f33-71bae9122857.png">

Reply:
> Removing the height and using flex makes the tabBarItem looks like this

It shouldn't be using `flex`. The main point is that the height should be based on what's inside the item without hardcoding, or if it's hardcoded to a specific height then it shouldn't be based on the height of tab bar at bottom. It needs to be its own thing based on what looks good in horizontal tab bar.

Reply:
Removing the flex property still returns the same result as flex: 0

I totally understand why we shouldn't user-defined heights cause when I increase the size of the icon with a defined height on the tabBarItem it seems to cut the tabBarItem text out.

Just not sure how to approach this problem.
But I am thinking of using layouts to calculate the children (icon and text) passed to the tabBarItem to get it's height and use that as the tabBarItemHeight this way we have dynamic height.
What do you think of this approach @satya164 ?

Reply:
> But I am thinking of using layouts to calculate the children (icon and text) passed to the tabBarItem to get it's height and use that as the tabBarItemHeight this way we have dynamic height.

Calculating layouts is asynchronous so it won't produce good results. As I wrote, if it can't be dynamic and is hardcoded then it needs to be its own value, not based on the bottom tab's height.